### PR TITLE
QField should register and deal with expired/invalidated tokens

### PR DESCRIPTION
**Pull request by nirvn**
@suricactus , as discussed earlier.

This PR handles all the UI/UX issues I stumbled with invalidated tokens.

### CODE REVIEW

**Comment by nirvn** [comment]
The biggest logic addition here is the auto-disconnect when the server tells us we are using an invalid token. This meant _not_ changing the cloud projects list in the main model as we don't want to shift the list index in mid air while we are trying to synchronize for e.g.

Instead, if the user goes to the cloud projects list while offline, we use our new filter proxy to hide those.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for fc04e191d4a14ba9f9b35bf0331535eaad2f300c:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-14eb43ef26fcfe87b1c0f74dd3c8b2f2bd290e08-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-14eb43ef26fcfe87b1c0f74dd3c8b2f2bd290e08-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-14eb43ef26fcfe87b1c0f74dd3c8b2f2bd290e08-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-14eb43ef26fcfe87b1c0f74dd3c8b2f2bd290e08-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 59091cf92e2fbc9139f4dd062c273b63a3fecc6f:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-a83c395c94a57a4493926d9f07945755560ba354-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-a83c395c94a57a4493926d9f07945755560ba354-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-a83c395c94a57a4493926d9f07945755560ba354-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1839-a83c395c94a57a4493926d9f07945755560ba354-x86.apk)

File: src/core/qfieldcloudprojectsmodel.cpp [hunk 1]
```diff
@@ -191,8 +191,8 @@ void QFieldCloudProjectsModel::refreshProjectsList()
     }
     case QFieldCloudConnection::ConnectionStatus::Disconnected:
     {
-      QJsonArray projects;
-      reload( projects );
+      // Nothing done at the moment; since the connection can be disconnected after discovering
+      // a token has been invalidated, be careful with what is added here.
```
Comment (suricactus) [inline src/core/qfieldcloudprojectsmodel.cpp:195]: So in the end we will show the last loaded projects list here?

**Review by suricactus — APPROVED** [review]
Looks good to me

**Pull request merged**